src/transport/SessionHandle.h

```diff
@@ -26,12 +26,14 @@ namespace Transport {
 class Session;
 } // namespace Transport
 
-class SessionHolder;
-
 /** @brief
- *    Non-copyable session reference. All SessionHandles are created within SessionManager. SessionHandle is not
- *    reference *counted, hence it is not allowed to store SessionHandle anywhere except for function arguments and
- *    return values. SessionHandle is short-lived as it is only available as stack variable, so it is never dangling. */
+ *    Non-copyable session reference. All SessionHandles are created within SessionManager. It is not allowed to store SessionHandle
+ * anywhere except for function arguments and return values.
+ *
+ *    SessionHandle is reference counted such that it never dangling, but there can be a gray period when the session is mark as
+ * pending removal but not actually removed yet. During this period, the handle is functional, but the underlying session won't be
+ * able to grabbed by any SessionHolder. SessionHandle->IsActiveSession can be used to check if the session is active.
```

Comment:
This comment referencing SessionHolder specific bits should be added to SessionHolder.h, not here.

Reply:
Unlike `SessionHandle`, An `SessoinHolder` can't point to a pending-to-removal session.